modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ConvertProcessor.java

```diff
@@ -86,8 +104,14 @@ public Object convert(Object value) {
                     return INTEGER.convert(value);
                 } catch (IllegalArgumentException e) {}
                 try {
+                    return LONG.convert(value);
+                } catch (IllegalArgumentException e) {}
+                try {
                     return FLOAT.convert(value);
                 } catch (IllegalArgumentException e) {}
+                try {
+                    return DOUBLE.convert(value);
```

Comment:
Problem here is that `FLOAT.convert(value)` would succeed and produce potential rounding errors instead of failing and dropping down to here.

Looks like setting `AUTO` would potentially never generate a FLOAT if we re-order these